.github/workflows/container-image.yml

```diff
@@ -39,202 +37,155 @@ jobs:
-      run: make OS_NAME=centos build-toolbox
-
 
   build-client:
+    strategy:
+      matrix:
+        os: [centos, fedora, opensuse]
+        arch: [amd64]
     runs-on: ubuntu-latest
-    env:
-      BUILDAH_FORMAT: oci
     steps:
     - uses: actions/checkout@v3
-    - name: Build the client image
-      run: make build-client
+    - name: build the client image
+      run: make OS_NAME=${{ matrix.os }} BUILD_ARCH=${{ matrix.arch }} build-client
     # Here we upload samba-client image to artifacts locally for consumption
-    # during samba-toolbox build process.
-    - name: Upload client image
+    # during  the samba-toolbox build process.
+    - name: Upload the client image
       uses: ishworkh/docker-image-artifact-upload@v1
       with:
-        image: "quay.io/samba.org/samba-client:latest"
+        image: "quay.io/samba.org/samba-client:${{ matrix.os }}-latest"
         retention_days: 1
 
   build-toolbox:
+    strategy:
+      matrix:
+        os: [centos, fedora, opensuse]
+        arch: [amd64]
     needs: build-client
     runs-on: ubuntu-latest
-    env:
-      BUILDAH_FORMAT: oci
     steps:
     - uses: actions/checkout@v3
     # Download locally stored samba-client image to be used as base for building
     # samba-toolbox.
     - name: Download client image
       uses: ishworkh/docker-image-artifact-download@v1
       with:
-        image: "quay.io/samba.org/samba-client:latest"
+        image: "quay.io/samba.org/samba-client:${{ matrix.os }}-latest"
     - name: Build the toolbox image
-      run: make build-toolbox
+      run: make OS_NAME=${{ matrix.os }}  BUILD_ARCH=${{ matrix.arch }} build-toolbox
 
   test-server:
+    strategy:
+      matrix:
+        os: [centos, fedora, opensuse]
+        arch: [amd64]
     needs: build-server
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
     - name: Download server image
       uses: ishworkh/docker-image-artifact-download@v1
       with:
-        image: "samba-server:latest"
+        image: "samba-server:${{ matrix.os }}-latest"
     - name: Test the server image
-      run: tests/test-samba-container.sh
+      run: LOCAL_TAG="samba-server:${{ matrix.os}}-latest" tests/test-samba-container.sh
 
   # Reminder: the nightly-server images consume nightly samba rpm builds
   # it is not *just* an image that gets built nightly
-  build-nightly-server:
-    runs-on: ubuntu-latest
-    env:
-      BUILDAH_FORMAT: oci
-    steps:
-    - uses: actions/checkout@v3
-    - name: Build the nightly server image
-      run: make build-nightly-server
-    - name: Upload nightly server image
-      uses: ishworkh/docker-image-artifact-upload@v1
-      with:
-        image: "samba-server:nightly"
-        retention_days: 1
-
-  build-nightly-server-centos:
+  build-nightly-servers:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [centos, fedora]
+        arch: [amd64]
+        exclude:
+          - os: centos
```

Comment:
Why do we exclude `centos` here? Nightly builds works well with CentOS in building server images. I think we should have a common `build-nightly-server` job for `centos` and `fedora` without any exclusion. We should also have a common `build-nightly-ad-server` job for `centos` and `fedora` excluding `centos` similar to `build-ad-server` job.

Reply:
@anoopcs9 , we had to exclude centos  from som jobs, specifically those building ad-server images, because the jobs failed as we seem to never never have committed the  corresponding containerfile to the master branch. I Agree that if possible, we should add centos, but it will require some extra work to make it for ad-server initially. I am quite sure that I had that in my original branch and PR to support building with different base OS, but it seems to never have landed

Reply:
Yes, that's exactly what I commented above. **build-nightly-server** will work for `centos` and **build-nightly-ad-server** won't currently work for `centos`. I've raised #142 to add Containerfile for CentOS to build ad-server image with nightly samba builds.